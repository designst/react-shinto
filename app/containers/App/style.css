.App {
  text-align: center;
}

.App-logo {
  height: 100px;
}

.App-header {
  color: #3a506b;
  height: 150px;
  padding: 20px;
  margin-bottom: 20px;
  background-color: #0b132b;
}

.App-title {
  font-size: 1.5em;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}
