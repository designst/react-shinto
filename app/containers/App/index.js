import React from 'react';
import { hot } from 'react-hot-loader';
import { Helmet } from 'react-helmet';
import { renderRoutes } from 'react-router-config';

import config from 'config';

import './style.css';
import logo from './logo.svg';

type Props = { route: Object };

class App extends React.Component {
  componentDidMount() {
    // Remove the server-side injected CSS.
    const jssStyles = document.getElementById('jss-server-side');

    /* istanbul ignore next */
    if (jssStyles && jssStyles.parentNode) {
      jssStyles.parentNode.removeChild(jssStyles);
    }
  }

  props: Props;

  render() {
    const { route } = this.props;

    return (
      <div className="App">
        <Helmet {...config.app} />

        <header className="App-header">
          <img src={logo} alt="logo" className="App-logo" />

          <h1 className="App-title">{config.app.title}</h1>
        </header>

        {renderRoutes(route.routes)}
      </div>
    );
  }
}

export default hot(module)(App);
