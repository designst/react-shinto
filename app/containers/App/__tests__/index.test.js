import React from 'react';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom';

import App from '../index';

describe('<App />', () => {
  test('renders', () => {
    const mockData = {
      route: {},
    };

    const tree = renderer
      .create(
        <StaticRouter context={{}}>
          <App {...mockData} />
        </StaticRouter>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
