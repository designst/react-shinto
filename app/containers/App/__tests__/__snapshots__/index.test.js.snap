// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.png"
    />
    <h1
      className="App-title"
    >
      React Shint√µ
    </h1>
  </header>
</div>
`;
